Tests/Rules/ReviewUnusedParameter.tests.ps1

```diff
@@ -5,63 +5,60 @@ Describe "ReviewUnusedParameter" {
     }
 
     Context "When there are violations" {
-        $Cases = @(
-            @{
-                ScriptDefinition   = 'function BadFunc1 { param ($Param1, $Param2) $Param1}'
-                Name               = "function with 1 unused parameter"
-                NumberOfViolations = 1
-            }
-            @{
-                ScriptDefinition   = 'function BadFunc1 { param ($Param1, $Param2) }'
-                Name               = "function with 2 unused parameters"
-                NumberOfViolations = 2
-            }
-            @{
-                ScriptDefinition   = '{ param ($Param1) }'
-                Name               = "scriptblock with unused parameter"
-                NumberOfViolations = 1
-            }
-            @{
-                ScriptDefinition   = '{ param ($Param1) }; { $Param1 }'
-                Name               = "parameter with same name in different scope"
-                NumberOfViolations = 1
-            }
-        )
-        It "has 1 violation for <Name>" -TestCases $Cases {
-            param ($ScriptDefinition, $Name, $NumberOfViolations)
+        It "has 1 violation - function with 1 unused parameter" {
```

Comment:
Oh, I just meant you can inline the tests like this:

```powershell
It "has no violations for function <Name>" -TestCases @( 
        @{
            ScriptDefinition = '...'
            Name = '...'
            NumberOfViolations = 1
       }
        @{
            ScriptDefinition = '...'
            Name = '...'
            NumberOfViolations = 1
       }
    ) {
        param ($ScriptDefinition, $Name)
        $Violations = Invoke-ScriptAnalyzer -ScriptDefinition $ScriptDefinition -IncludeRule $RuleName
        $Violations.Count | Should -Be $NumberOfViolations
        # etc.
    }
```

Reply:
I see. Still, it looks like most of the other test files are not using test cases this way, so I figured I'd go with the flow. The bonus is that the test names are a little clearer this way I think. Your call...

Reply:
Happy for you to leave it 🙂, just meant to express that I didn't mean for you to go to all the trouble of unfolding the tests